hello.c

```diff
@@ -1,46 +1,59 @@
+#include <stdbool.h>
 #include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
-#include <assert.h>
+#include <string.h>
 
-#include "launchdarkly/api.h"
+#include <launchdarkly/api.h>
 
-#define YOUR_SDK_KEY "<put your SDK key here>"
-#define YOUR_FEATURE_KEY "<put your feature key here>"
+// Set SDK_KEY to your LaunchDarkly SDK key.
+#define SDK_KEY ""
+
+// Set FEATURE_FLAG_KEY to the feature flag key you want to evaluate.
+#define FEATURE_FLAG_KEY "my-boolean-flag"
+
+#define INIT_TIMEOUT_MILLISECONDS 3000
+
+int main() {    
+    if (!strlen(SDK_KEY)) {
+        printf("*** Please edit hello.c to set SDK_KEY to your LaunchDarkly SDK key first\n\n");
+        return 1;
+    }
 
-int
-main()
-{
     struct LDConfig *config;
     struct LDClient *client;
     struct LDUser *user;
 
-    setbuf(stdout, NULL);
+    LDConfigureGlobalLogger(LD_LOG_INFO, LDBasicLogger);
 
-    LDConfigureGlobalLogger(LD_LOG_TRACE, LDBasicLogger);
-    LDGlobalInit();
+    config = LDConfigNew(SDK_KEY);
 
-    config = LDConfigNew(YOUR_SDK_KEY);
-    assert(config);
+    // Set up the user properties. This user should appear on your LaunchDarkly users dashboard
+    // soon after you run the demo.
+    user = LDUserNew("example-user-key");
+    LDUserSetName(user, "Sandy");
```

Comment:
In `hello-c-client`, the user was configured before initializing the SDK because that's how the client-side SDKs work. But in the server-side SDKs, there's no need to do that and I think it'd be slightly clearer (and more representative of what an application would really do) to initialize the SDK first, and then configure the user. See for instance [hello-go](https://github.com/launchdarkly/hello-go/blob/master/main.go#L26).

Reply:
Oh actually the spot above `LDClientInit` is a bad idea anyways because of the early exit when the client is not initialized.